VultisigApp/VultisigApp/Chains/cosmos/cosmos.swift

```diff
@@ -254,4 +243,46 @@ class CosmosHelper {
             throw HelperError.runtimeError("fail to get signed transaction,error:\(error.localizedDescription)")
         }
     }
+    
+    private func buildCosmosFee(gas: UInt64) -> CosmosFee {
+        return CosmosFee.with {
+            $0.gas = gasLimit
+            $0.amounts = [CosmosAmount.with {
+                $0.denom = denom
+                $0.amount = String(gas)
+            }]
+        }
+    }
+    
+    private func buildCosmosWasmGenericMsg(keysignPayload: KeysignPayload) throws -> CosmosMessage {
+        let coinType = keysignPayload.coin.chain.coinType
+        
+        guard coinType.validate(address: keysignPayload.coin.address) else {
+            throw HelperError.runtimeError("Invalid Address type: \(keysignPayload.coin.address)")
+        }
+        
+        guard let contractPayload = keysignPayload.wasmExecuteContractPayload else {
+            throw HelperError.runtimeError("Invalid empty WasmExecuteContractPayload")
+        }
+        
+        let formattedMessage = contractPayload.executeMsg
+            .replacingOccurrences(of: "^\\{", with: "{ ", options: .regularExpression)
+            .replacingOccurrences(of: "\\}$", with: " }", options: .regularExpression)
+            .replacingOccurrences(of: ":", with: ": ")
+        let coins = contractPayload.coins.map { coin in
```

Comment:
this is pending to removed on Android @gastonm5 . Extension will format dapps payload on their end